### PR TITLE
I can activate activities when entering an ad-hoc subprocess

### LINKED ISSUE #25271: I can activate activities when entering an ad-hoc subprocess
**Description**

I want to activate activities when the process instance enters an ad-hoc subprocess. 

- Add a new Zeebe extension element `adhoc` with the attribute `activateElements`.
- The attribute `activateElements` should be a FEEL expression that returns a list of BPMN element ids that should be activated.
- When the process instance activates the ad-hoc subprocess, it evaluates the attribute.
  - If the value is a list of strings, the process instance activates the BPMN elements whose element ids are in the list.
  - If the value is not a list of strings, the process instance creates an incident. 
  - If the value is a list of strings and contains other values than the BPMN element ids, the process instance creates an incident. 
- If the attribute is not set, the process instance activates no activities but just the ad-hoc subprocess.    

Requires https://github.com/camunda/camunda/issues/25270.

### PR DESCRIPTION
**Pull request by saig0**
## Description

- Add a new BPMN extension element `adHoc` for ad-hoc subprocesses with one attribute `activeElementsCollection` (type: FEEL expression)
- Evaluate the expression when activating an ad-hoc subprocess. If the evaluation result is valid, it activates the BPMN child elements of the ad-hoc subprocess. If the result is not valid, it creates an incident.
- When the ad-hoc subprocess is terminated, the process instance terminates all child instances of the ad-hoc subprocess.

## Related issues

closes #25271

### CODE REVIEW

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java [hunk 1]
```diff
@@ -248,8 +256,19 @@ public void shouldApplyInputMappings() {
...
-            tuple("x", "1", processInstanceKey),
-            tuple("local_x", "2", adHocSubProcessActivated.getKey()));
+            tuple("activateElements", "[\"A\"]", processInstanceKey),
+            tuple("activateElements", "[\"A\",\"B\"]", adHocSubProcessActivated.getKey()));
+
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(AD_HOC_SUB_PROCESS_ELEMENT_ID, ProcessInstanceIntent.ELEMENT_ACTIVATED))
+        .extracting(r -> r.getValue().getElementId(), Record::getIntent)
```
Comment (ana-vinogradova-camunda) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java:266]: 💭 For the future contributors: may be something like `describedAs` would be useful to explain what order is expected here.

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java [hunk 2]
```diff
@@ -133,8 +256,19 @@ public void shouldApplyInputMappings() {
...
+
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit("B", ProcessInstanceIntent.ELEMENT_ACTIVATED))
+        .extracting(r -> r.getValue().getElementId(), Record::getIntent)
+        .containsSequence(
+            tuple(AD_HOC_SUB_PROCESS_ELEMENT_ID, ProcessInstanceIntent.ELEMENT_ACTIVATING),
+            tuple(AD_HOC_SUB_PROCESS_ELEMENT_ID, ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple("A", ProcessInstanceIntent.ACTIVATE_ELEMENT),
```
Comment (ana-vinogradova-camunda) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java:270]: 💭 For the future contributors: may be something like short describedAs would be useful to explain what order is expected here.

**Review by ana-vinogradova-camunda — APPROVED** [review]
Great PR 🚀 

👍 Well structured and small commits make it easy to review the changes
👍 Test coverage and tests themselves are also very helpful

Just one 💭  - should some `describedAs` be added for the tests in order to improve tests readability?

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java (hunk 2 above)
Comment (saig0) [reply zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/AdHocSubProcessTest.java:270]: In general, I agree that `describedAs(..)` is useful to provide additional context in a test. 

In this specific case, I don't see the need for it. The expected order is clearly defined by the assertion `containsSequence(..)`.

**Pull request merged**